robotpy_build/command/build_pyi.py

```diff
@@ -159,6 +140,68 @@ def main():
         cfg["out"],
     ] + cfg["stubs"]
 
+    math_to_add = {}
+
+    class fp_wrapper:
+        def __init__(self, fp) -> None:
+            self.fp = fp
+        
+        def write(self, s: str) -> None:
+            # fix underscored names
+            s = re.sub(r"([a-zA-Z0-9_]+)\._\1", r"\1", s)
+            self.fp.write(s)
+
+        def __getattr__(self, __name: str) -> Any:
+            return getattr(self.fp, __name)
+
+
+    @contextmanager
+    def open_wrapper(file, *args, **kwargs):
+        with open(file, *args, **kwargs) as fp:
+            import typing
+            fp: typing.TextIO
+            ### before pyi write
+            fp.write("#\n# AUTOMATICALLY GENERATED FILE, DO NOT EDIT!\n#\n\n")
+
+            # yield fp ### pyi write
+            yield fp_wrapper(fp) ### pyi write
+            ### after pyi write
+
+            # Add buffer definition
+            if sys.version_info >= (3, 12):
+                fp.write("\n\nfrom collections.abc import Buffer\n\n")
+            else:
+                fp.write("\n\nBuffer = typing.Union[bytes, bytearray, memoryview]\n\n")
```

Comment:
This seems like something that should be contributed upstream, rather than us monkeypatching it in.

Reply:
It is an existing issue: https://github.com/sizmailov/pybind11-stubgen/issues/87

But right now, I don't actually check if the type hints use `buffer` or if the type hint will override another custom type called buffer. That bit gets injected into every file. I would want to make sure the buffer stuff is only added when necessary+valid before upstreaming.